### PR TITLE
Auto cleanup scabbard pruning

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR enables auto cleanup of pruned scabbard state trees by default, or allows the feature to be disabled by the use of the flag `--disable-scabbard-autocleanup`

### CODE REVIEW

**Review by vaporos — DISMISSED** [review]

File: splinterd/packaging/splinterd.toml.example [hunk 1]
```diff
@@ -31,6 +31,10 @@ version = "1"
 # files will be created in the Splinter state_dir.
 #scabbard_state = "database"
 
+# Enable Auto-cleanup of Scabbard state;o
```
Comment (agunde406) [inline splinterd/packaging/splinterd.toml.example:34]: typeo ";o"

File: services/scabbard/libscabbard/src/service/factory/mod.rs [hunk 2]
```diff
@@ -164,6 +164,7 @@ pub struct ScabbardFactoryBuilder {
     state_storage_configuration: Option<ScabbardLmdbStateConfiguration>,
     storage_configuration: Option<ScabbardStorageConfiguration>,
     signature_verifier_factory: Option<Arc<Mutex<Box<dyn VerifierFactory>>>>,
+    enable_state_autocleanup: Option<bool>,
```
Comment (agunde406) [inline services/scabbard/libscabbard/src/service/factory/mod.rs:167]: typo in commit message: ads -> adds

**Review by agunde406 — CHANGES_REQUESTED** [review]

**Review by agunde406 — APPROVED** [review]

File: splinterd/packaging/splinterd.toml.example (hunk 1 above)
Comment (peterschwarz) [reply splinterd/packaging/splinterd.toml.example]: Fixed

File: services/scabbard/libscabbard/src/service/factory/mod.rs (hunk 2 above)
Comment (peterschwarz) [reply services/scabbard/libscabbard/src/service/factory/mod.rs:167]: Fixed

**Review by vaporos — APPROVED** [review]

**Pull request merged**